services/outputhost/extcache.go

```diff
@@ -306,6 +330,68 @@ func (extCache *extentCache) loadReplicaStream(startAddress int64, startSequence
 	return
 }
 
+func (extCache *extentCache) loadKafkaStream(
+	cgName string,
+	outputHostUUID string,
+	startFrom common.UnixNanoTime,
+	kafkaCluster string,
+	kafkaTopics []string,
+) (repl *replicaConnection, err error) {
+	groupID := getKafkaGroupIDForCheramiConsumerGroupName(cgName)
+
+	// Configure sarama-cluster
+	cfg := sc.NewConfig()
+
+	// Metadata for the Kafka group join
+	meta := KafkaGroupMetadata{
+		Version:        kafkaGroupMetadataVersion,
+		OutputHostUUID: outputHostUUID,
+	}
+	cfg.Group.Member.UserData, _ = json.Marshal(meta)
+
+	// Get the notifications channel; we will just log it
+	cfg.Group.Return.Notifications = true
+
+	// Older startFroms (e.g. 0, >1 weeks back) are considered to want the oldest offset
+	// TODO: Use Sarama GetMetadata to get the list of partitions, then build the offset request
+	// to use with GetAvailableOffsets, and then "somehow" manually commit it so that sarama-cluster
+	// starts from the right place
+	if common.Now()-startFrom > common.UnixNanoTime(time.Hour*24*7) {
```

Comment:
When the startFrom is less than 1 week, will it use lastest offset? If yes, feel a little risky here, since users may not be aware of that they are actually consuming from latest even though they specify a "start from" time. Like the TODO here. And feel we'd better default to oldest offset before we implement that TODO.